04.ls/ls.rb

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+opt = OptionParser.new
+opt.parse!(ARGV)
+
+def directory_contents
+  directory = '.'
+  Dir.children(directory).reject { |file| file.start_with?('.') }.sort
+end
+
+def display_in_columns(files, max_columns = 3)
+  max_length = files.map(&:length).max || 0
+  column_width = max_length + 2
+
+  rows = (files.size.to_f / max_columns).ceil
+  rows.times do |row|
+    line = []
+    max_columns.times do |col|
+      index = row + col * rows
+      line << (files[index] || '').ljust(column_width)
+    end
```

Comment:
これでも悪くはないのですが、
```suggestion
    line = 
      max_columns.times.map do |col|
        index = row + col * rows
        (files[index] || '').ljust(column_width)
      end
```
と `map` などもうまく使えるようになるとよりプログラミングの幅が広がると思います。

Reply:
コメントありがとうございます。
とても勉強になります🙇‍♀️

Dirクラスについてもう少しマニュアルを読んで、理解を深めておきます。
`map`を使えることに気付けていませんでした。
`map`も今後使いこなせるようにしていきたいと思います。